src/steps/user/index.ts

```
@@ -42,28 +42,33 @@ export async function buildUserGroupUserRelationships({
   instance,
   jobState,
 }: IntegrationStepExecutionContext<SonarqubeIntegrationConfig>) {
+  const client = createSonarqubeClient(instance.config);
   await jobState.iterateEntities(
     { _type: Entities.USER._type },
     async (userEntity) => {
-      const client = createSonarqubeClient(instance.config);
-      await client.iterateUsersGroups(async (userGroup) => {
-        const userGroupEntityId = createUserGroupEntityIdentifier(userGroup.id);
-        const userGroupEntity = await jobState.findEntity(userGroupEntityId);
-
-        if (!userGroupEntity) {
-          throw new IntegrationMissingKeyError(
-            `Expected user group with key to exist (key=${userGroupEntityId})`,
+      await client.iterateGroupsAssignedToUser(
+        userEntity.login as string,
+        async (userGroup) => {
+          const userGroupEntityId = createUserGroupEntityIdentifier(
+            userGroup.id,
           );
-        }
+          const userGroupEntity = await jobState.findEntity(userGroupEntityId);
 
-        await jobState.addRelationship(
-          createDirectRelationship({
-            _class: RelationshipClass.HAS,
-            from: userGroupEntity,
-            to: userEntity,
-          }),
-        );
-      }, userEntity.login as string);
+          if (!userGroupEntity) {
+            throw new IntegrationMissingKeyError(
+              `Expected user group with key to exist (key=${userGroupEntityId})`,
+            );
+          }
```

Now's probably a good time to mention that you'll probably see some integrations that throw if `jobState.findEntity` doesn't find an entity, and others that just silently continue. Keep this in mind as you work through integrations because it will be great to gain a consensus on the right approach over time.